### DISCUSSION TITLE
Forward model luminosities

### DISCUSSION THREAD

**Discussion by jsinkbaek** [phoebe-project/phoebe2#546] [Questions]
Hello!

I am normally working with JKTEBOP, where I normalize my light curve using polynomial curve fitting around the eclipses ahead of analysis. I wish to ascertain what effect this has on the stellar radii and other parameters, so I decided to produce a forward model using PHOEBE and my found JKTEBOP parameters, and normalize it in the same way before using it as a synthetic light curve in JKTEBOP. However, I want to be absolutely sure that I'm handling it right, since I'm getting some wild results.

In order to get the right luminosity ratio in the forward model without accurate temperatures, I am currently using the *decoupled* setting in *pblum_mode*, and then manually scaling the secondary component *pblum* using my previously found luminosity ratio. Is that reasonable?

What about third light? If I manually input something like 
*b['l3_frac@lc_kepler@dataset'] = 0.0029667000*
will that still be included in the forward model?

I suspect that I might be doing something wrong somewhere. It is possible that everything is as it's supposed to be. I'm just a little bit skeptical since I'm seeing a 60%-90% increase in radius for both my components by doing this, which I suspect might be a bit extreme. Is it possible that the eclipse depths are simply too deep in the forward model due to some setting I'm using?

The change in eclipse depths for the forward model is also visible in an over-plot:
![phoebe_total](https://user-images.githubusercontent.com/56873820/138507457-2c539133-38fe-4f2b-b3e1-43793cc69c61.png)
![phoebe_annular](https://user-images.githubusercontent.com/56873820/138507460-5cf8dfb4-d5d6-4c14-a05e-bdce7deeae1d.png)

I have also attached a pdf version of my notebook, if that is helpful.
[forward.pdf](https://github.com/phoebe-project/phoebe2/files/7400695/forward.pdf)

**Comment by kecnry** [comment]
The third light should definitely be in the forward model, so I don't think that should be a problem, unless its defined at a different reference time/flux which is possible.  

I would caution against using decoupled luminosities unless absolutely necessary (as luminosities will no longer be coupled with radii and temperatures).  Instead it might be worth trying to back out the temperatures that give you the luminosity ratio (probably surface brightness ratio) from jktebop, and see if that agrees better.  You can find the way that we estimate surface brightness ratio from phoebe parameters when calling jktebop in the [jktebop wrapper](https://github.com/phoebe-project/phoebe2/blob/master/phoebe/backend/backends.py#L2120).

My last suggestion would be to just run the forward model with PHOEBE's jktebop wrapper and compare those to see how well the code's compare in this area of parameter space (keeping in mind some discrepancies could also be caused by approximations in the wrapper itself... including this ugly surface brightness approximation).

**Reply by jsinkbaek** [reply]
What do you mean by the third light being defined at a different reference time/flux?

It sounds like the easiest solution, if I don't want to try to work out some kind of custom constraints, is to do some form of minimization of the calculated surface brightness ratio in relation to the JKTEBOP ones. I'll try to do that, and report back.

**Reply by jsinkbaek** [reply]
It seemed like the surface brightness trick did not produce better results, so I've stuck to the decoupled pblums. To give some idea of why I think that's still reasonable: The stellar light ratio fitted to my original light curve is much better determined than the radii and temperatures, given that I have a total eclipse present. Therefore I'm more comfortable with using that to produce a forward model than the two other parameters.

The following two changes seem to have fixed the issue, since I'm getting much more reasonable results now:

- I changed the limb darkening in PHOEBE to be a manual quadratic law using the same coefficients as in JKTEBOP (maybe it makes more sense to keep the interpolated values, but since that's not what I'm trying to quantify I wanted as little difference as possible between the models as possible, except for deformation and irradiation).
- I got JKTEBOP to iteratively calculate errors for my data points. It seems that the program does not handle light curves without errors in any reasonable fashion. I find it odd that it's even an option then.

Thank you very much for taking the time to assist me!

**Reply by kecnry** [reply]
Glad to hear you got reasonable results, thanks for the updates!

> What do you mean by the third light being defined at a different reference time/flux?

Since you're providing third light in fractional units, those need to be converted to a constant third light flux based on the total system flux, which is of course time-dependent in a binary system.  Each code that supports fractional third light likely has different assumptions on how to make this conversion.

**Comment by jsinkbaek** [comment]
Hello @kecnry 

This is an old thread, but I have a follow-up question.
Is it possible to change the phase that pblums are defined at?

I wish to adjust so that it is defined at some-place close to eclipses, instead of at maximum light or periastron (which is current?)

Example plot where luminosity~1 at maximum light:
![image](https://user-images.githubusercontent.com/56873820/165272265-f6fe2317-ef3f-46c7-b764-95ddcd9aeb0a.png)

**Reply by aprsa** [reply]
@jsinkbaek, passband luminosity is an _aspect-independent_ quantity; it has no awareness of the phase or eclipses. It is per-star power output in the specific passband at time t0. The only reason why it depends on time is because there may be time-variable phenomena that change the luminosity. Flux, on the other hand, is clearly a phase-dependent quantity that is linearly proportional to passband luminosities, which might be the root of confusion.

**Reply by jsinkbaek** [reply]
@aprsa thank you for the info. However, if I change t0 I do not observe any change in the model when defining the same pblum as before?

Follow-up: Is third light also handled at t0?

**Reply by aprsa** [reply]
You would only ever expect to see a change if a system luminosity is temporally variable (i.e. by anything that changes the power output of the star, such as pulsations). All parameter values that can change in time are by convention given at t0. For example, if orbital period is variable (dpdt != 0), then the passed orbital period will correspond to the instantaneous period at time t0.

Is your goal to normalize the light curve at some particular phase for a given passband?

**Reply by jsinkbaek** [reply]
Yes, my goal is to compare bi-axial ellipsoid models of JKTEBOP with those of PHOEBE given that I have a specific measurement of third light and luminosity ratio at a phase and passband.

**Reply by jsinkbaek** [reply]
This is not for a minimization, but for a visual comparison of the two models using the same, previously found, solution.

**Reply by aprsa** [reply]
I understand. This is tricky business as luminosity scaling is done differently in both codes and not easily reconcilable -- @kecnry can elaborate on that because he looked into other codes more thoroughly than me. For comparison purposes I would advocate for post-run scaling at a certain phase (understanding, of course, that you are scaling _flux_ and not luminosity).

**Reply by kecnry** [reply]
@jsinkbaek - you are correct that the pblum computation/scaling is done at `t0@system` (but not `t0_perpass`, `t0_supconj`, etc), but the luminosity isn't usually strongly phase-dependent (although can be for eccentric systems in addition to things like pulsations that Andrej mentioned but aren't supported yet).  You can see the phase-dependence of the luminosity by adding a mesh dataset and exposing the `pblum_ext` column (plotting these in phoebe is quite clunky, so you might want to plot them manually or just look at the numbers).  If that shows a significant phase-dependence but changing `t0` doesn't shift the fluxes appropriately, please send me a short snippet to reproduce if you can and I'll look into it (I don't think I've ever tested it directly, but would expect the fluxes to change if the luminosities do).

If third light is _fractional_, then that will be computed to the additive flux value at `t0@system` as well (based on the flux at time `t0`).

I agree with @aprsa that you're probably better off manually re-scaling if you want to compare models between different codes (unless of course you want/need to compare the scaling itself).  The logic to attempt to reconcile the scaling between jktebop and phoebe (within phoebe's jktebop wrapper) is explained in Section 3.4.4 of the [fitting release paper](http://phoebe-project.org/publications/2020Conroy+).... but is not perfect as you can see in [Figure 6 which compares two models across the various backend wrappers](http://phoebe-project.org/docs/latest/examples/backends_compare_legacy_jktebop_ellc).  If you want more details to either reproduce this logic outside phoebe or if you want to attempt to improve it, let me know and I'd be happy to point you to where the logic lives in the code.

**Reply by jsinkbaek** [reply]
So please correct me if I am wrong, but so far my understanding is:

- pblums are normally ~unchanged and all (most?) of deformation and reflection is incorporated only in the pbflux along with other geometric effects (eclipses).

- Providing fractional third light gives a static flux addition relative to pbflux at t0. Meaning it depends on which phase it is provided in. If I define t0@system = t0_supconj (within primary eclipse), would this mean that flux amount of third light is significantly lower for the whole model, compared with providing it at f.ex. maximum flux or some other t0?

- If I, with t0 = t0_supconj and no third light, give pblums for both components such that total flux is 1 if no eclipse was present and stars were spherical, e.g. pblum_A = 4pi/(1+L_B/L_A)   and   pblum_B = 4pi * (1 - 1/(1+L_B/L_A)), then the model should have flux ~1 at around primary eclipse (if the eclipse is ignored).

The last effect is the one I am most puzzled by, because that is what I tried to do for the plot above where it seems that flux (sorry for the confusion with luminosity before I see it now) is ~1 at maximum flux, not near eclipse. I have managed to rescale it fine outside of PHOEBE. But I was just confused since it did not behave like I expected when I changed t0@system.

**Reply by jsinkbaek** [reply]
Reading back, I realise now that at what t0@system I define pblums should probably not have an effect on what pbflux is calculated in my case. I assume this means pblum provided this way always relates to the maximum flux?

Is there a way that I can work backwards? E.g. calculate pblums given that I know what the flux of both components should be at a specific phase?

**Reply by aprsa** [reply]
In essence, `pblum` is the value assigned to the integrated model atmosphere-dependent emergent passband intensity from the entire star at time `t0`. If the star was spherical and unobscured, you would then get flux from that star that is close to but not exactly `pblum/4pi`, and that is because of the model atmospheres -- specific passband intensity variations compared to blackbody make this relationship not exact. Thus, there is no ready way to request `pbflux@t0=value` as we'd need to iterate on `pblum` to get there. Under the hood we calculate _absolute_ `pblum` from model atmospheres (that is internal to phoebe and unexposed unless you change `pblum_mode`) and calculate a scaling factor that turns it into the parameterized `pblum` (i.e., assigned by you). This sets the "units" for the entire radiative description in phoebe.

**Reply by kecnry** [reply]
Just wanted to add a clarification about third light.

> If I define t0@system = t0_supconj (within primary eclipse), would this mean that flux amount of third light is significantly lower for the whole model, compared with providing it at f.ex. maximum flux or some other t0?

My explanation overly simplified this, so sorry for the confusion.  Under-the-hood third light is always added as a constant flux at all phases.  If you provide third light in fractional units (`l3_mode='fraction'`), then this constant flux is estimated ignoring distortion and eclipses (by using the expected `pblum/4pi`).  You can always see how that mapping will re-act by calling [b.compute_l3s](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.compute_l3s) manually.  Unless you're forced to work in fractional third light, its probably best to just work with flux units.

**Reply by jsinkbaek** [reply]
Ok thank you both. I think that gives me an overall idea about it.

@kecnry that makes sense! I was also a bit confused why it would be a good idea to have it distortion and eclipse dependent.
I have not been able to properly resolve the differences between l3_frac and JKTEBOP third light, so I have decided to simply use a light curve that has previously been corrected for third light.
